WIP: V8 Debugging with SourceLink (#1769)
umbraco/UmbracoDocs: Reference/Debugging/index.md

---
versionFrom: 8.1.0
---

> # Debugging

Reviewer: wondering whether we should call the article 'Debugging with SourceLink' ?

we have a page called 'debugging' already, in getting started - code - debugging...

https://our.umbraco.com/documentation/Getting-Started/Code/Debugging/

and we could add at the footer of that page a link back to this reference page... ?

Author: ```suggestion
# Debugging with SourceLink
```

> In version Umbraco 8.1+ we have enabled a debugging feature from Microsoft & Visual Studio called SourceLink which allows you to step into the native Umbraco CMS source code.

Reviewer: Wondering if this is better ?

Microsoft & Visual Studio have introduced a new debugging technology called 'SourceLink' that enables source code debugging of certain .NET assemblies from NuGet - in version Umbraco 8.1+ this feature has been enabled to allow developers to step into the native Umbraco CMS source code.

Author: ```suggestion
Microsoft & Visual Studio have introduced a new debugging technology called 'SourceLink' that enables source code debugging of certain .NET assemblies from NuGet - in version Umbraco 8.1+ this feature has been enabled to allow developers to step into the native Umbraco CMS source code.
```

> ## Enabling SourceLink in Visual Studio 2019

Reviewer: Is this the same in vs2017?

Author: I need to check & verify @marcemarc 90% sure it is

Author: ```suggestion
## Enabling SourceLink in Visual Studio 2017 & 2019
```

* Verify that VS Debugging option **SourceLink** checked & **Just My Code** option is unchecked

![Visual Studio 2019 Debug Settings for SourceLink](Images/VS19-enable-sourcelink.png)

## What is SourceLink ?
You can find details from Microsoft about SourceLink here:

> https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink<br/>

Reviewer: maybe these are better in a bulleted list - with MD links saying where they go... 

eg

- Microsoft Docs
- Scott Hanselman blog post on Sourcelink
- SourceLink on GitHub

https://www.hanselman.com/blog/ExploringNETCoresSourceLinkSteppingIntoTheSourceCodeOfNuGetPackagesYouDontOwn.aspx<br/>
https://github.com/dotnet/sourcelink

> ## Verifying this works

Reviewer: instead of verifying this works how about:

Working with SourceLink

Author: ```suggestion
## Working with SourceLink
```

* Create new 4.7.2 .NET Framework blank/empty website

> * Install latest UmbracoCMS 8.1+ Nuget Packages from Nuget.org

Reviewer: will this only work if Umbraco is installed via Nuget, apologies I've only skim read  the links...

Author: Yes as the vanilla ZIP package the bin folder does not contain the PDBs which has the magic SourceLink meta data inside them

* Create an IUserComposer or similar code in your new site/SLN that you want to F11/Step Into

> * Prompt should appear & the original source code file is fetched directly from GitHub

Reviewer: can we add a screengrab of prompt?

Author: Yep good idea 👍

> * How far can you `F11` aka `Step Into` & go down the rabbit hole of source code?

Reviewer: go down the rabbit hole of the Umbraco CMS source code?

Author: ```suggestion
* How far can you `F11` aka `Step Into` & go down the rabbit hole of the Umbraco CMS source code?
```

---
versionFrom: 8.1.0
---

# Debugging with SourceLink

> Microsoft & Visual Studio have introduced a new debugging technology called 'SourceLink' that enables source code debugging of certain .NET assemblies from NuGet - in version Umbraco 8.1+ this feature has been enabled to allow developers to step into the native Umbraco CMS source code.

Reviewer 2: ```suggestion
Microsoft & Visual Studio have introduced a new debugging technology called 'SourceLink' that enables source code debugging of certain .NET assemblies from NuGet. In version Umbraco 8.1+ this feature has been enabled to allow developers to step into the native Umbraco CMS source code.
```
